### PR TITLE
Add block-number along with validation code for a given TXID

### PR DESCRIPTION
**Pull request by manish-sethi**
Signed-off-by: manish <manish.sethi@gmail.com>

#### Type of change
- Improvement (improvement to code, performance, etc)

#### Description
This allows gateway to pass on additional relevant info to client and for using it for logging that could be helpful in debugging

### CODE REVIEW

**Review by bestbeforetoday — APPROVED** [review]

File: common/ledger/blkstorage/blockstore.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ func (store *BlockStore) RetrieveBlockByTxID(txID string) (*common.Block, error)
 }
 
 // RetrieveTxValidationCodeByTxID returns the validation code for the specified txID
-func (store *BlockStore) RetrieveTxValidationCodeByTxID(txID string) (peer.TxValidationCode, error) {
+func (store *BlockStore) RetrieveTxValidationCodeByTxID(txID string) (peer.TxValidationCode, uint64, error) {
```
Comment (mastersingh24) [inline common/ledger/blkstorage/blockstore.go:96]: This is a change to a public method, but given Fabric is not designed to be a library, I don't think we need to document the API change.

File: core/committer/txvalidator/v14/validator_test.go [hunk 2]
```diff
@@ -1450,9 +1450,9 @@ func (m *mockLedger) GetBlockByTxID(txID string) (*common.Block, error) {
 }
 
 // GetTxValidationCodeByTxID returns validation code of give tx
-func (m *mockLedger) GetTxValidationCodeByTxID(txID string) (peer.TxValidationCode, error) {
+func (m *mockLedger) GetTxValidationCodeByTxID(txID string) (peer.TxValidationCode, uint64, error) {
```
Comment (mastersingh24) [inline core/committer/txvalidator/v14/validator_test.go:1453]: This is a change to a public method, but given Fabric is not designed to be a library, I don't think we need to document the API change.

**Review by mastersingh24 — APPROVED** [review]
One comment about a single-use function, but other than that change seems fine to me

File: common/ledger/blkstorage/blockstore.go (hunk 1 above)
Comment (manish-sethi) [reply common/ledger/blkstorage/blockstore.go:96]: Good Catch! I'll fix this in a separate PR to avoid delay merging in this.

**Pull request merged**